mono/metadata/marshal.c

```diff
@@ -117,8 +117,14 @@ get_method_image (MonoMethod *method)
 	return m_class_get_image (method->klass);
 }
 
+// func is an identifier, that names a function, and is also in jit-icall-reg.h,
+// and therefore a field in mono_jit_icall_info and can be token pasted into an enum value.
+//
+// The name of func must be linkable for AOT, for example g_free does not work (monoeg_g_free instead),
+// nor does the C++ overload fmod (mono_fmod instead). These functions therefore
+// must be extern "C".
 #define register_icall(func, sig, no_wrapper) \
```

Comment:
Not sure this block of comments need to be repeated multiple times.

Reply:
Agreed it is annoying and not sure what to do. 

If they were identical, move to header, but they aren’t.

And what to document vs. read the code.

People follow contradictory styles — literate programming where comments are like a book and no need to read the code, or code is eminently readable and comments are for the rare hard to understand code, to be avoided.